summingbird-storm/src/main/scala/com/twitter/summingbird/storm/BaseBolt.scala

```diff
@@ -149,12 +150,13 @@ case class BaseBolt[I, O](jobID: JobId,
     }
   }
 
-  private def finish(inputs: Stream[InputState[Tuple]], results: TraversableOnce[O]) {
+  private def finish(inputs: Chain[InputState[Tuple]], results: TraversableOnce[O]) {
+    val tuples = inputs.iterator.map(_.state).toList
```

Comment:
shouldn't we move this line to 158 (only materialize the `List` if we have dependants and we anchor?